src/eckit/stats/Comparator.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+#include <iosfwd>
+#include <string>
+
+
+namespace eckit {
+class Parametrisation;
+namespace stats {
+class Field;
+}
+}  // namespace eckit
+
+
+namespace eckit::stats {
+
+
+class Comparator {
+public:
+    // -- Constructors
+
+    Comparator(const Parametrisation&, const Parametrisation&);
+    Comparator(const Comparator&) = delete;
+
+    // -- Destructor
+
+    virtual ~Comparator();
+
+    // -- Operators
+
+    void operator=(const Comparator&) = delete;
+
+    // -- Methods
+
+    virtual std::string execute(const Field&, const Field&) = 0;
```

Comment:
std::string is a very non-obvious return type for a Comparator. Unless it is doing something weird.

Reply:
Indeed this sholuld change -- it is a list of reasons why a comparison failed, and if empty is it a successful comparison. I should bring in more (a lot more) semantics here. Good catch!